config/resolve_credentials.go

```diff
@@ -151,6 +156,20 @@ func resolveCredsFromProfile(ctx context.Context, cfg *aws.Config, envConfig *En
 	return nil
 }
 
+func resolveSSOCredentials(ctx context.Context, cfg *aws.Config, sharedConfig *SharedConfig, configs configs) error {
+	var options []func(*ssocreds.Options)
+	v, found, err := getSSOProviderOptions(ctx, configs)
+	if err != nil {
+		return err
+	}
+	if found {
+		options = append(options, v)
+	}
+
+	cfg.Credentials = ssocreds.New(sso.NewFromConfig(cfg.Copy()), sharedConfig.SSOAccountID, sharedConfig.SSORegion, sharedConfig.SSORoleName, sharedConfig.SSOStartURL, options...)
```

Comment:
Do the SDK's client's `NewFromConfig` functions not already perform a copy internally on the `Config` type?  If not they probably should all be doing that internally. Because if not any modifications to the config's list or pointer member values could pollute the client if the passed in `Config` value is later modified.

Reply:
Yes they do copy already, will remove this unnecessary step.